morepath/dispatch.py

```diff
@@ -0,0 +1,73 @@
+import reg
+from functools import wraps, partial
+from .reify import reify
+
+
+class delegate(object):
+
+    def __init__(self, *predicates):
+        self.make_generic = reg.dispatch(*predicates)
+        self.external_predicates = False
+
+    def __call__(self, func):
+
+        def install(reg):
+            setattr(reg, func.__name__, self.make_generic(func))
+
+        @wraps(func)
+        @reify
```

Comment:
I see now reify is used here to make this calculation only once. So after the initial call the dispatch is under a partial function invocation.